cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/table/field/FieldValidator.java

```diff
@@ -59,6 +60,35 @@ public void validateField(Field<?> field) throws InvalidFieldException {
     }
   }
 
+  /**
+   * Validate if the given range matches the schema. The given range is invalid if:
+   * its field is not present in the given schema,
+   * its field type is different from the given schema,
+   * if its field is a primary key but the given value is null,
+   * or it does not contain a primary key.
+   *
+   * @param range the range to validate
+   * @throws InvalidFieldException if the field does not pass the validation
+   */
+  public void validateRange(Range range) throws InvalidFieldException {
+    range.getBegin().forEach(this::validateField);
+    range.getEnd().forEach(this::validateField);
+
+    if (hasNoPrimaryKey(range.getBegin()) && hasNoPrimaryKey(range.getEnd())) {
```

Comment:
Just check that the first field in filter is first field in PK. I am not sure how our queries would work (especially for NoSQL) if you do something like `[non-pk:{start1, end1}, pkfield2:{start2, end2}]` So, we should always have first pk field first in range

Reply:
Note: I see that you have this check for NoSQL, but it's better to have checks consistent between the backends otherwise it's hard to test and easy to miss something

Reply:
got it, updated